packages/vkui/src/components/AppRoot/ScrollContext.tsx

```diff
@@ -86,30 +86,69 @@ function useScrollLockController(enableScrollLock: () => void, disableScrollLock
   };
 }
 
+export function useScrollingLockedScroll(): Pick<ScrollContextInterface, 'scrollTo' | 'getScroll'> {
+  const { scrollTo, getScroll } = React.useContext(ScrollContext);
+  return React.useMemo(
+    () => ({
+      scrollTo,
+      getScroll,
+    }),
+    [getScroll, scrollTo],
+  );
+}
+
 export interface ScrollControllerProps extends HasChildren {
   elRef: React.RefObject<HTMLElement | null>;
 }
 
 export const GlobalScrollController = ({ children }: ScrollControllerProps): React.ReactNode => {
   const { window, document } = useDOM();
   const beforeScrollLockFnSetRef = React.useRef<Set<() => void>>(new Set());
+  const scrollLockEnabledRef = React.useRef(false);
 
   const getScroll = React.useCallback<ScrollContextInterface['getScroll']>(
-    (options = { compensateKeyboardHeight: true }) => ({
-      x: window!.pageXOffset,
-      y: options.compensateKeyboardHeight
-        ? getPageYOffsetWithoutKeyboardHeight(window!)
-        : window!.pageYOffset,
-    }),
-    [window],
+    (options = { compensateKeyboardHeight: true }) => {
+      if (!window) {
+        throw new Error('window is not defined');
```

Comment:
```suggestion
      if (!window) {
        throw new Error('window is not defined');
```
давай лучше возвращать пустые данные `{ x: 0, y: 0 }`

в данном случае выбрасывание ошибки слишком жестко